src/telegram.js

```diff
@@ -491,11 +491,13 @@ class TelegramBot extends EventEmitter {
 
     if (message) {
       debug('Process Update message %j', message);
-      this.emit('message', message);
       const processMessageType = messageType => {
         if (message[messageType]) {
           debug('Emitting %s: %j', messageType, message);
           this.emit(messageType, message);
+          this.emit('message', { ...message, type: messageType });
+        } else {
+          this.emit('message', message);
```

Comment:
I see a bug! Considering that the function `processMessageType` in invoked several times (below: `TelegramBot.messageTypes.forEach(processMessageType);`), this branch will cause the `message` event to be emitted in any case where the message is of the type being inspected!

Considering that a message is of **exactly** one type e.g. `audio`, `text`, etc, we could first find the type. For example,

```js
const messageType = TelegramBot.messageTypes.find((type) => message[type]);
message.type = messageType;
```
Then we can now emit the events:

```js
this.emit("message", message);
this.emit(messageType, message);
```
Using `.find()` will actually improve our performance.